include/deal.II/base/aligned_vector.h

```diff
@@ -463,7 +469,26 @@ class AlignedVector
   BOOST_SERIALIZATION_SPLIT_MEMBER()
 #endif
 
+  /**
+   * Exception message for changing the vector after a call to
+   * replicate_across_communicator().
+   *
+   * @ingroup Exceptions
+   */
+  DeclExceptionMsg(ExcAlignedVectorChangeAfterReplication,
+                   "Changing the vector after a call to "
+                   "replicate_across_communicator() is not allowed.");
```

Comment:
I have introduced this exception message which we can use for all non-const functions in a follow-up.